src/wallet/rpcdump.cpp

```diff
@@ -323,6 +323,103 @@ RPCHelpMan importaddress()
     };
 }
 
+static void RemoveAddress(CWallet*, const CTxDestination&, bool);
+static void RemoveScript(CWallet* const pwallet, const CScript& script, bool isRedeemScript, bool purgeTxs) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
+{
+    isminetype ismine = pwallet->IsMine(script);
+    if (!isRedeemScript && ismine == ISMINE_SPENDABLE) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "The wallet contains the private key for this address");
+    } else if (!isRedeemScript && ismine == ISMINE_NO) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "The wallet does not contain this address or script");
+    }
+    CHECK_NONFATAL(ismine == ISMINE_WATCH_ONLY || isRedeemScript);
+
+    LegacyScriptPubKeyMan& spk_man = EnsureLegacyScriptPubKeyMan(*pwallet);
+
+    if (spk_man.HaveWatchOnly(script) && !spk_man.PurgeWatchOnly(script)) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Error removing address/script from wallet");
+    }
+    if (isRedeemScript) {
+        CScriptID scriptID = CScriptID(script);
+        if (!spk_man.HaveCScript(scriptID)) {
+            throw JSONRPCError(RPC_WALLET_ERROR, "Error removing p2sh redeemScript from wallet");
+        }
+        RemoveAddress(pwallet, ScriptHash(scriptID), purgeTxs);
+    } else {
+        CTxDestination destination;
+        if (ExtractDestination(script, destination)) {
+            pwallet->DelAddressBook(destination);
+        }
+    }
+
+    if (purgeTxs) {
+        pwallet->RemoveTransactions(script);
+    }
+}
+
+static void RemoveAddress(CWallet* const pwallet, const CTxDestination& dest, bool purgeTxs) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
+{
+    CScript script = GetScriptForDestination(dest);
+    RemoveScript(pwallet, script, false, purgeTxs);
+    // remove from address book
+    if (IsValidDestination(dest)) {
+        pwallet->DelAddressBook(dest);
+    }
+}
+
+RPCHelpMan removeaddress()
+{
+    return RPCHelpMan{
+        "removeaddress",
+        "\nRemoves an address or script (in hex) that was being watched as if it were in your wallet but was not being used to spend. Requires a new wallet backup.\n",
+        {
+            {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The Bitcoin address (or hex-encoded script)"},
+            {"p2sh", RPCArg::Type::BOOL, /* default */ "false", "Remove the P2SH version of the script as well"},
+            {"purgetransactions", RPCArg::Type::BOOL, /* default_val */ "false", "Remove existing transactions from the wallet"},
```

Comment:
ISTM current practice is to use snake case for the field name
```suggestion
            {"purge_transactions", RPCArg::Type::BOOL, /* default_val */ "false", "Whether to remove existing transactions from the wallet"},
```